### PR TITLE
Edit Readme.md for grammar and fluency

### PR DESCRIPTION
**Pull request by ChaseBro**
I tried to simplify where possible while still retaining the flavor, let me know if something doesn't sound right.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,60 +3,62 @@ Dotfiler — the ultimate solution for managing dotfiles!
...
+2. Clone some config files into the `$HOME/.dotfiles`.
 3. Run `dot update` to make all necessary symlinks.
 4. Have a profit!
 
-How does it work
-----------------
+Overview
+------------
+
+From user's point of view — very simply. You just create a separate subdirectory called "environments", put configs there and then run `dot update`. Dotfiler will make all necessary symlinks automagickally. **What makes dotfiler better, than other solutions?** It's ability to merge files from different environments into one target dir. Here's an example:
```
Comment (cybtachyon) [inline README.md]: KISS, just like this project. Remove **_From user's point of view... ... You just_.**

Replace with **_Create_** a separate subdirectory...

Fix **What makes dotfiler _better than_ other solutions? _Its_ ability to merge..**

File: README.md [hunk 2]
```diff
@@ -3,60 +3,62 @@ Dotfiler — the ultimate solution for managing dotfiles!
...
+
+Suppose, you have a `~/.zshrc` which sources all configs from `~/.zsh/`. And you want to separate default configs from the configs only needed on work machines. In most config managers you will end up with two separate repositories, each sharing part of zsh config. But dotfiler allows you to do a much more clever thing — separate zsh (actually any other configs too, if they understand `include`) into the different environments.
```
Comment (cybtachyon) [inline README.md]: This whole paragraph should probably be rewritten. It has too many commas and illogical separation of clauses.

File: README.md [hunk 3]
```diff
@@ -3,60 +3,62 @@ Dotfiler — the ultimate solution for managing dotfiles!
...
+You can also add new environments using `dot add <url> <url>...`. (Probably the process of adding environments on a fresh machine will be even more improved, when I introduce a 'meta-environments', which will allow you to make one env depend on other envs and pull them automatically when adding)
 
 Get involved
 ------------
 
-Don't hesitate to try dotfiler. Just install it and make your configs more structured.  Extract useful ones and share them in the GitHub, as I did. Then send me a link with a short description (or make a pull request), and I'll add you repositories to the end of this page. 
+Don't hesitate to try dotfiler. Just install it and make your configs more structured.  Extract useful ones and share them on GitHub, as I did. Then send me a link with a short description (or make a pull request), and I'll add you repositories to the end of this page.
```
Comment (cybtachyon) [inline README.md]: **add _your_ repositories to the**

File: README.md [hunk 4]
```diff
@@ -3,60 +3,62 @@ Dotfiler — the ultimate solution for managing dotfiles!
...
 
-Dotfiler was developed in TDD, it's core functionality is fully tested, but that doesn't mean there isn't bugs. If you have found one, file the issue, or better, try to write a test for the use case, fix it and send as a pull request. To run all tests, install nose and run `nosetests bin/lib/dot`. 
+Dotfiler's core functionality is fully tested, but that doesn't mean there aren't bugs. If you find one, file the issue on Github, or even better, try to write a test and/or fix for that use case and send it as a pull request. To run all tests, install nose and run `nosetests bin/lib/dot`.
```
Comment (cybtachyon) [inline README.md]: Add an example command for installing `node`.

File: README.md [hunk 5]
```diff
@@ -3,60 +3,62 @@ Dotfiler — the ultimate solution for managing dotfiles!
...
 
-More technical details
-----------------------
-
-If you are wondering, how does dotfiler work inside, I'll tell you. 
+How it works
+------------
 
-First of all, it walks through all files and all environments collecting all dirs, mentioned in more than one environment and files. If file with same filename exits in few environments, this is an error and `dot` will tell you they are conflicting. 
+First dotfiler, walks through all files and all environments collecting dirs and files mentioned in more than one environment as a tree. If a file with same filename exits in more than one environment this is an error and `dot` will tell you they are conflicting.
```
Comment (cybtachyon) [inline README.md]: **First _dotfiler walks_**

**same filename _exists_ in more than**

File: README.md [hunk 6]
```diff
@@ -168,13 +170,10 @@ LINK    Symlink from /home/art/.zsh/ssh-agent to /home/art/.dotfiles/zsh/.zsh/ss
...
 ```
 
-Have you got the idea? Good! File an issue or (better) send a pull-request.
-
 How to ignore some files
 ========================
 
-Edit a config file `~/.dotfiles/.dotignore` and add any regex
-patterns which you need.
+Edit a config file `~/.dotfiles/.dotignore` and add any regex patterns you need.
```
Comment (cybtachyon) [inline README.md]: Add a small example.

**Comment by svetlyak40wt** [comment]
Thank you!.

**Comment by cybtachyon** [comment]
@svetlyak40wt  Would you prefer a second pull request with those suggested fixes, or @ChaseBro do you plan to address the feedback given?

**Comment by svetlyak40wt** [comment]
Sure.

**Pull request merged**